File: c++/Easy/Part001/Fibonacci Sequence.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cin >> n;

    if (n < 3 || n > 50) {
        return 1;
    }

    long long a = 0, b = 1;
    cout << a << " ";

    if (n > 1) {
        cout << b << " ";
    }

    for (int i = 3; i <= n; i++) {
        long long next = a + b;
        cout << next << " ";
        a = b;
        b = next;
    }

    return 0;
}

File: Java/Easy/Part001/Fibonacci Sequence.java
import java.util.Scanner;

public class Hello {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        if (n < 3 || n > 50) {
            return;
        }

        long a = 0, b = 1;
        System.out.print(a + " ");

        if (n > 1) {
            System.out.print(b + " ");
        }

        for (int i = 3; i <= n; i++) {
            long next = a + b;
            System.out.print(next + " ");
            a = b;
            b = next;
        }

        scanner.close();
    }
}
